packages/flutter_tools/test/general.shard/dap/flutter_adapter_test.dart

```diff
@@ -214,6 +214,35 @@ void main() {
       });
     });
 
+    group('handles reverse requests', () {
+      test('app.exposeUrl', () async {
+        final MockFlutterDebugAdapter adapter = MockFlutterDebugAdapter(
+          fileSystem: MemoryFileSystem.test(style: fsStyle),
+          platform: platform,
+        );
+
+        // Pretend to be the client, handling any reverse-requests for exposeUrl
+        // and mapping the host to 'mapped-host'.
+        adapter.exposeUrlHandler = (String url) => Uri.parse(url).replace(host: 'mapped-host').toString();
+
+        // Simulate Flutter asking for a URL to be exposed.
+        const int requestId = 12345;
+        adapter.simulateStdoutMessage(<String, Object?>{
+          'id': requestId,
+          'method': 'app.exposeUrl',
+          'params': <String, Object?>{
+            'url': 'http://localhost:123/',
+          }
+        });
+
+        // Allow the handler to be processed.
+        await pumpEventQueue(times: 5000);
```

Comment:
cool, didn't know about `pumpEventQueue()`. However, why 5000 times?

Reply:
It was pretty arbitrary.. The default is like 20, but I was worried that might be racy if the handler code happens to `await` a few times. The analysis server tests use 5000 in a lot of places so I've just always assumed that's a reasonable number (and that the cost of this is very low but the cost of investigating flakes might not be).

I don't know if this is a good reason though, so I'm happy to change - or even remove if makes sense.

Reply:
Ahh, ok. My main concern was that this code path had 4983 asynchronous suspensions, and a later refactor might break this test. This is fine.